helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -117,6 +122,16 @@ private WagedRebalancer(AssignmentMetadataStore assignmentMetadataStore,
     _rebalanceAlgorithm = algorithm;
     _mappingCalculator = mappingCalculator;
     _manager = manager;
+    // If metricCollector is null, instantiate a version that does not register metrics in order to
+    // allow rebalancer to proceed
+    _metricCollector =
+        metricCollector == null ? new WagedRebalancerMetricCollector() : metricCollector;
+  }
+
+  @VisibleForTesting
```

Comment:
As you suggested in the other PR, it is not preferred to use "@VisibleForTesting" method, right? Shall we remove this annotation and move it closer to the other protected method?